src/lib/homeobject_impl.hpp

```diff
@@ -43,26 +55,30 @@ class HomeObjectImpl : public HomeObject,
     auto _defer() const { return folly::makeSemiFuture().via(folly::getGlobalCPUExecutor()); }
 
 protected:
+    std::mutex _repl_lock;
+    std::shared_ptr< home_replication::ReplicationService > _repl_svc;
+    //std::shared_ptr<homestore::ReplicationService> _repl_svc;  
     peer_id _our_id;
 
     /// Our SvcId retrieval and SvcId->IP mapping
     std::weak_ptr< HomeObjectApplication > _application;
 
     ///
     mutable std::shared_mutex _pg_lock;
-    using shard_set = std::unordered_set< ShardInfo >;
-    using pg_pair = std::pair< PGInfo, shard_set >;
-    std::map< pg_id, pg_pair > _pg_map;
+    std::map< pg_id, PG > _pg_map;
 
     mutable std::shared_mutex _shard_lock;
-    std::map< shard_id, shard_set::const_iterator > _shard_map;
+    std::map < shard_id, ShardPtr > _shard_map;
```

Comment:
This can be `ShardList::const_iterator`.

Reply:
how about ShardList::Iterator?  because seal shard will change the status of shard info, so it can not be const, right?

Reply:
Yes `iterator` is fine; i just tend to make things more `const` if possible first and weaken the type if it doesn't work. You can always convert a const_iterator to non-const with:
```auto it = container.erase(const_it, const_it);```
but it looks odd in the code.